### PR TITLE
Enabling better logging and catching important errors

### PR DESCRIPTION
**Pull request by oudeismetis**
- [x] Rotate the log files
- [x] Confirm that all messages are getting logged correctly (seeing some issues)

### CODE REVIEW

**Comment by oudeismetis** [comment]
This PR is not 100% ready yet (see items in the list). But it's basically done, so comments are welcome.

File: franklin/builder/models.py [hunk 1]
```diff
@@ -34,14 +34,28 @@ def build(self):
...
-            print('Response HTTP Response Body : {content}'.format(
-                content=r.content))
+        r = None
+        try:
+            r = requests.post(url, data=json.dumps(body), headers=headers)
+        except (ConnectionError, HTTPError, Timeout) as e:
+            logger.error('Response HTTP Status Code   : %s', e.status_code)
+            logger.error('Response HTTP Response Body : %s', e.content)
+        except:
+            logger.error('Unexpected Builder error: %s', sys.exc_info()[0])
```
Comment (oudeismetis) [inline franklin/builder/models.py]: Apparently %s is the most `pythonic` way of constructing these types of logs. http://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html

**Comment by JulianGindi** [comment]
This all looks good. Did you test the try/catch with the whole system? If so, go ahead and rebase into master.

**Comment by oudeismetis** [comment]
I did test it, but by pointing my local api to production builder. Having the same issues cornelius is having. We should definitely re-run end-to-end testing once I merge to make sure we don't have any lingering fragility from last night.

**Pull request closed without merging**